augmentation/augmentation.hpp

```diff
@@ -170,7 +213,119 @@ class Augmentation
     }
   }
 
-  //! Locally held augmentations and transforms that need to be applied.
+/**
+   * Function to determine if augmentation has horizontal-flip function.
+   *
+   * @param augmentation Optional argument to check if a string has
+   *                     horizontal-flip substring.
+   */
+  bool HasHorizontalFlipParam(const std::string& augmentation = "")
+  {
+    if (augmentation.length())
+      return augmentation.find("horizontal-flip") != std::string::npos;
+
+
+    // Search in augmentation vector.
+    for(size_t i=0; i<augmentations.size(); i++)
+    {
+      if(augmentations[i].find("horizontal-flip") != std::string::npos)
+          return true;
+    }
+    return false;
+
+  }
+
+  /**
+   * Function to determine if augmentation has vertical-flip function.
+   *
+   * @param augmentation Optional argument to check if a string has
+   *                     vertical-flip substring.
+   */
+  bool HasVerticalFlipParam(const std::string& augmentation = "")
+  {
+    if (augmentation.length())
+      return augmentation.find("vertical-flip") != std::string::npos;
+
+
+    // Search in augmentation vector.
+    for(size_t i=0; i<augmentations.size(); i++)
+    {
+      if(augmentations[i].find("vertical-flip") != std::string::npos)
+          return true;
+    }
+    return false;
+
+  }
+
+  /**
+   * find if new data should horizontal flipped.
+   *
+   * @param ishortiflip Output is horizontal flipped or not.
+   * @param augmentation String from boolean value is extracted.
+   */
+  void GetHorizontalFlipParam(bool& ishortiflip,
+                      const std::string& augmentation)
+  {
+    if (!HasHorizontalFlipParam())
+      return;
+
+    ishortiflip = false;
+
+    // Use regex to find true or false.
+    boost::regex regex{"(?:true|false)"};
+
+    // Create an iterator to find matches.
+    boost::sregex_token_iterator matches(augmentation.begin(),
+        augmentation.end(), regex, 0), end;
+
+    size_t matchesCount = std::distance(matches, end);
+
+    if (matchesCount == 1)
+    {
+      ishortiflip = (*matches) == "true" ? true:false;
+    }
+    else
+    {
+        mlpack::Log::Fatal << "Invalid boolean value in " <<
+          augmentation << std::endl;
+    }
+  }
```

Comment:
If we remove the need for true and false, assume we need to flip if there is horizontal flip then we can remove this as well.

Reply:
yes